arcgis-ios-sdk-samples/Edit data/Edit features with feature-linked annotation/EditFeaturesWithFeatureLinkedAnnotationViewController.swift

```diff
@@ -0,0 +1,228 @@
+        // Clear any previously selected features.
+        clearSelection()
+        selectedFeature = nil
+        
+        // Identify across all layers.
+        mapView.identifyLayers(atScreenPoint: screenPoint, tolerance: 10.0, returnPopupsOnly: false) { (results: [AGSIdentifyLayerResult]?, error: Error?) in
+            if let error = error {
+                self.presentAlert(error: error)
+                return
+            }
+            results?.forEach { (result) in
+                //Get a reference to the identified feature layer and feature.
+                guard let featureLayer = result.layerContent as? AGSFeatureLayer, let selectedFeature = result.geoElements[0] as? AGSFeature else { return }
+                switch selectedFeature.geometry?.geometryType {
+                case .point:
+                    // If the selected feature is a point, prompt the edit attributes alert.
+                    self.showEditableAttributes(selectedFeature: selectedFeature)
+                case .polyline:
+                    let polyline = selectedFeature.geometry as! AGSPolyline
+                    let polylineArray = polyline.parts.array()
+                    // If the selected feature is a polyline with any part containing more than one segment (i.e. a curve), present an alert.
+                    for part in polylineArray {
+                        if part.pointCount > 2 {
+                            // Set the value to nil.
+                            self.selectedFeature = nil
+                            // Present the alert.
+                            self.presentAlert(title: "Make a different selection", message: "Select straight (single segment) polylines only.")
+                            return
+                        }
+                    }
+                    // If the selected feature is a valid polyline, set the value to true.
+                    self.selectedFeatureIsPolyline = true
+                default:
+                    return
+                }
+                // Select the feature.
+                featureLayer.select(selectedFeature)
+                // Set the feature globally.
+                self.selectedFeature = selectedFeature
+            }
+        }
+    }
+    
+    // Create an alert dialog with edit texts to allow editing of the given feature's 'AD_ADDRESS' and 'ST_STR_NAM' attributes.
+    func showEditableAttributes(selectedFeature: AGSFeature) {
+        // Create an alert controller and customize the title and message.
+        let alert = UIAlertController(title: "Edit feature attributes", message: "Edit the 'AD_ADDRESS' and 'ST_STR_NAM' attributes.", preferredStyle: .alert)
+        // Add text fields to prompt user input.
+        alert.addTextField()
+        alert.addTextField()
+        // Populate the text fields with the current values.
+        alert.textFields?[0].text = (selectedFeature.attributes["AD_ADDRESS"] as! NSNumber).stringValue
```

Comment:
Can `attributes["AD_ADDRESS"]` be `nil`? If so, the force cast would be cause a crash in that case.

Reply:
No, it can't be nil.